src/effect.cpp

```diff
@@ -407,6 +408,11 @@ bool effect_type::load_decay_msgs(JsonObject &jo, std::string member)
 
 std::string effect::disp_name() const
 {
+    if (eff_type->name.size() < 1) {
```

Comment:
> Sure. But the bigger question is still: do we even't want this kind of guards?

As much as possible should be tested when the effect data is loaded. Currently `intensity` can only be 1, 2 or 3 (is that actually correct?), so the load function should check that `names` contains either nothing or exactly 3 entries. Then one only needs to check for emptiness before accessing `name[...]`. If an empty vector is not allowed at all, that should be checked (and rejected) upon load as well (and not checked here).

The code is a bit inconsistent about whether `name` can be empty. `get_speed_name` checks for an empty vector, `disp_name` does not and assumes it always non-empty. I don't know whether `name` should be allowed to be empty. If empty is not allowed, the check for it in `get_speed_name` should be removed (it's superfluous).


In short: if it's conceptually required (as in: `names` *must* have exactly three entries), it should be enforced when loading the data. Everything else should assume this requirement is already checked.

Similar with intensity. If there is an invariant (intensity must be 1, 2 or 3), it should be enforced whenever the value is changed (make sure it's changed to a valid value). Other code should simply assume it's correct. Clamping in `decay` is certainly correct.